addons/mail/static/src/models/autocomplete-input-view/autocomplete-input-view.js

```diff
@@ -0,0 +1,294 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one2one } from '@mail/model/model_field';
+import { clear } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'AutocompleteInputView',
+    identifyingFields: [['chatWindowOwnerAsNewMessageFormInput', 'discussAsMobileAddItemHeader', 'discussSidebarCategoryAsItemNewInput']],
+    recordMethods: {
+        doFocus() {
+            if (!this.component) {
+                return;
+            }
+            this.component.root.el.focus();
+        },
+        /**
+         * @param {MouseEvent} ev
+         */
+        onBlur(ev) {
+            this._hide();
+        },
+        onFocusin() {
+            if (this.chatWindowOwnerAsNewMessageFormInput) {
+                this.chatWindowOwnerAsNewMessageFormInput.update({ isFocused: true });
+            }
+        },
```

Comment:
It would be better if `isFocused` field was here, and chat window made a compute to it (either new message: is focus from input, or thread: is focus from composer)